singlestoredb/management/job.py

```diff
@@ -0,0 +1,844 @@
+    ) -> Job:
+        """Creates and returns a scheduled notebook job."""
+        if self._manager is None:
+            raise ManagementError(msg='JobsManager not initialized')
+
+        schedule = dict(
+            mode=mode.value,
+        )  # type: Dict[str, Any]
+
+        if start_at is not None:
+            schedule['startAt'] = from_datetime(start_at)
+
+        if execution_interval_in_minutes is not None:
+            schedule['executionIntervalInMinutes'] = execution_interval_in_minutes
+
+        execution_config = dict(
+            createSnapshot=create_snapshot,
+            notebookPath=notebook_path,
+        )  # type: Dict[str, Any]
+
+        if runtime_name is not None:
+            execution_config['runtimeName'] = runtime_name
+
+        if parameters is not None:
+            execution_config['parameters'] = [
+                dict(
+                    name=p[0],
+                    value=str(p[1]),
+                    type=ParameterType.from_str(type(p[1]).__name__),
+                ) for p in parameters
+            ]
+            print(execution_config['parameters'])
+
+        target_config = None  # type: Optional[Dict[str, Any]]
+        database_name = get_database_name()
+        if database_name is not None:
+            target_config = dict(
+                databaseName=database_name,
+            )
+
+            if resume_target is not None:
+                target_config['resumeTarget'] = resume_target
+
+            workspace_id = get_workspace_id()
+            virtual_workspace_id = get_virtual_workspace_id()
+            cluster_id = get_cluster_id()
+            if virtual_workspace_id is not None:
+                target_config['targetID'] = virtual_workspace_id
+                target_config['targetType'] = TargetType.VIRTUAL_WORKSPACE.value
+
+            elif workspace_id is not None:
+                target_config['targetID'] = workspace_id
+                target_config['targetType'] = TargetType.WORKSPACE.value
+
+            elif cluster_id is not None:
+                target_config['targetID'] = cluster_id
+                target_config['targetType'] = TargetType.CLUSTER.value
+
+        job_run_json = dict(
+            schedule=schedule,
+            executionConfig=execution_config,
+        )  # type: Dict[str, Any]
+
+        if target_config is not None:
+            job_run_json['targetConfig'] = target_config
+
+        if name is not None:
+            job_run_json['name'] = name
+
+        if description is not None:
+            job_run_json['description'] = description
+
+        res = self._manager._post('jobs', json=job_run_json).json()
+        return Job.from_dict(res, self)
+
+    def run(
+        self,
+        notebook_path: str,
+        runtime_name: Optional[str] = None,
+        parameters: Optional[List[Tuple[str, Any]]] = None,
+    ) -> Job:
+        """
+        Creates and returns a scheduled notebook job that
```

Comment:
Should have a single line summary followed by full description. This docstring also looks truncated.